### ISSUE TITLE
真人训练时时间巨长，8小时跑2%，而之前训练二次元同素材量、相似配置也就不到24小时跑完

### ISSUE THREAD

**Issue by testworksizan** [Akegarasu/lora-scripts#355]
时间过长，配置唯一变量就是dim由32，32变更为128，64，clip层数2调整为1。回调后速度没有明显改善，显卡驱动、cudnn、cuda均已安装。素材多为512*512（类似尺寸），大模型为realvisxlV40
下面是训练数据

使用目录内的 python 进行启动....
22:13:32-999508 INFO     Starting SD-Trainer Mikazuki GUI...
22:13:32-999508 INFO     Base directory: C:\Users\songz\Documents\app\lora-scripts-v1.7.3, Working directory:
                         C:\Users\songz\Documents\app\lora-scripts-v1.7.3
22:13:32-999508 INFO     Windows Python 3.10.11 C:\Users\songz\Documents\app\lora-scripts-v1.7.3\python\python.exe
22:13:32-999508 INFO     detected locale zh_CN, use pip mirrors
22:13:33-015191 INFO     Starting tageditor...
22:13:33-015191 INFO     Starting tensorboard...
22:13:33-140917 INFO     Server started at http://127.0.0.1:28000
22:13:36-808599 INFO     Torch 2.0.0+cu118
22:13:36-839753 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
22:13:36-839753 INFO     Torch detected GPU: NVIDIA GeForce RTX 4060 Ti VRAM 16380 Arch (8, 9) Cores 34
TensorBoard 2.10.1 at http://127.0.0.1:6006/ (Press CTRL+C to quit)
22:26:37-695791 INFO     Training started with config file / 训练开始，使用配置文件:
                         C:\Users\songz\Documents\app\lora-scripts-v1.7.3\config\autosave\20240219-222637.toml
22:26:37-698333 INFO     Task 216a33f8-632e-4af0-bd02-aa8d5d198060 created
Loading settings from C:\Users\songz\Documents\app\lora-scripts-v1.7.3\config\autosave\20240219-222637.toml...
C:\Users\songz\Documents\app\lora-scripts-v1.7.3\config\autosave\20240219-222637
prepare tokenizers
update token length: 255
Using DreamBooth method.
prepare images.
found directory C:\Users\songz\Documents\app\lora-scripts-v1.7.3\train\testhuman\30_testhuman contains 83 image files
2490 train images with repeating.
0 reg images.
no regularization images / 正則化画像が見つかりませんでした
[Dataset 0]
  batch_size: 2
  resolution: (512, 768)
  enable_bucket: True
  min_bucket_reso: 256
  max_bucket_reso: 2048
  bucket_reso_steps: 32
  bucket_no_upscale: False

  [Subset 0 of Dataset 0]
    image_dir: "C:\Users\songz\Documents\app\lora-scripts-v1.7.3\train\testhuman\30_testhuman"
    image_count: 83
    num_repeats: 30
    shuffle_caption: True
    keep_tokens: 0
    keep_tokens_separator:
    caption_dropout_rate: 0.0
    caption_dropout_every_n_epoches: 0
    caption_tag_dropout_rate: 0.0
    caption_prefix: None
    caption_suffix: None
    color_aug: False
    flip_aug: False
    face_crop_aug_range: None
    random_crop: False
    token_warmup_min: 1,
    token_warmup_step: 0,
    is_reg: False
    class_tokens: testhuman
    caption_extension: .txt

[Dataset 0]
loading image sizes.
100%|████████████████████████████████████████████████████████████████████████████████| 83/83 [00:00<00:00, 1774.15it/s]
make buckets
number of images (including repeats) / 各bucketの画像枚数（繰り返し回数を含む）
bucket 0: resolution (416, 896), count: 30
bucket 1: resolution (448, 832), count: 90
bucket 2: resolution (480, 800), count: 90
bucket 3: resolution (512, 736), count: 60
bucket 4: resolution (512, 768), count: 600
bucket 5: resolution (544, 704), count: 390
bucket 6: resolution (576, 672), count: 270
bucket 7: resolution (608, 608), count: 240
bucket 8: resolution (608, 640), count: 30
bucket 9: resolution (640, 608), count: 60
bucket 10: resolution (672, 576), count: 180
bucket 11: resolution (704, 544), count: 180
bucket 12: resolution (736, 512), count: 90
bucket 13: resolution (768, 512), count: 120
bucket 14: resolution (800, 480), count: 30
bucket 15: resolution (832, 448), count: 30
mean ar error (without repeats): 0.018484023653886444
preparing accelerator
loading model for process 0/1
load StableDiffusion checkpoint: C:/Users/songz/Documents/app/lora-scripts-v1.7.3/sd-models/realvisxlV40_v40Bakedvae.safetensors
building U-Net
loading U-Net from checkpoint
U-Net:  <All keys matched successfully>
building text encoders
loading text encoders from checkpoint
text encoder 1: <All keys matched successfully>
text encoder 2: <All keys matched successfully>
building VAE
loading VAE from checkpoint
VAE: <All keys matched successfully>
Enable xformers for U-Net
import network module: networks.lora
[Dataset 0]
caching latents.
checking cache validity...
100%|████████████████████████████████████████████████████████████████████████████████| 83/83 [00:00<00:00, 5223.92it/s]
caching latents...
100%|██████████████████████████████████████████████████████████████████████████████████| 83/83 [00:11<00:00,  7.11it/s]
create LoRA network. base dim (rank): 128, alpha: 64
neuron dropout: p=None, rank dropout: p=None, module dropout: p=None
create LoRA for Text Encoder 1:
create LoRA for Text Encoder 2:
create LoRA for Text Encoder: 264 modules.
create LoRA for U-Net: 722 modules.
enable LoRA for text encoder
enable LoRA for U-Net
prepare optimizer, data loader etc.
bin C:\Users\songz\Documents\app\lora-scripts-v1.7.3\python\lib\site-packages\bitsandbytes\libbitsandbytes_cuda118.dll
use 8-bit AdamW optimizer | {}
override steps. steps for 20 epochs is / 指定エポックまでのステップ数: 24900
enable full bf16 training.
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 2490
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 1245
  num epochs / epoch数: 20
  batch size per device / バッチサイズ: 2
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 24900
steps:   0%|                                                                                 | 0/24900 [00:00<?, ?it/s]
epoch 1/20
steps:   2%|█                                               | 542/24900 [9:39:48<434:17:24, 64.19s/it, avr_loss=0.0977]
model_train_type = "sdxl-lora"
pretrained_model_name_or_path = "C:/Users/songz/Documents/app/lora-scripts-v1.7.3/sd-models/realvisxlV40_v40Bakedvae.safetensors"
v2 = false
train_data_dir = "C:/Users/songz/Documents/app/lora-scripts-v1.7.3/train/testhuman"
prior_loss_weight = 1
resolution = "512,768"
enable_bucket = true
min_bucket_reso = 256
max_bucket_reso = 2048
bucket_reso_steps = 32
output_name = "testhuman2"
output_dir = "./output"
save_model_as = "safetensors"
save_precision = "bf16"
save_every_n_epochs = 4
max_train_epochs = 20
train_batch_size = 2
gradient_checkpointing = false
network_train_unet_only = false
network_train_text_encoder_only = false
learning_rate = 0.0001
unet_lr = 0.0001
text_encoder_lr = 0.00001
lr_scheduler = "cosine"
lr_warmup_steps = 0
optimizer_type = "AdamW8bit"
min_snr_gamma = 5
network_module = "networks.lora"
network_dim = 128
network_alpha = 64
log_with = "tensorboard"
logging_dir = "./logs"
caption_extension = ".txt"
shuffle_caption = true
keep_tokens = 0
max_token_length = 255
seed = 1337
mixed_precision = "bf16"
full_bf16 = true
xformers = true
lowram = false
cache_latents = true
cache_latents_to_disk = true
persistent_data_loader_workers = true
